src/_algopy_testing/serialize.py

```diff
@@ -25,66 +25,71 @@ def identity(i: _T) -> _T:
-        return _Serializer(
-            native_type=typ,
-            arc4_type=arc4.Address,
-            native_to_arc4=arc4.Address,
-            arc4_to_native=lambda a: a.native,
-        )
-    if issubclass(typ, UInt64):
-        return _Serializer(
-            native_type=typ,
-            arc4_type=arc4.UInt64,
-            native_to_arc4=arc4.UInt64,
-            arc4_to_native=lambda a: a.native,
-        )
-    if issubclass(typ, Bytes):
-        return _Serializer(
-            native_type=typ,
-            arc4_type=arc4.DynamicBytes,
-            native_to_arc4=arc4.DynamicBytes,
-            arc4_to_native=lambda a: a.native,
-        )
-    if issubclass(typ, String):
-        return _Serializer(
-            native_type=typ,
-            arc4_type=arc4.String,
-            native_to_arc4=arc4.String,
-            arc4_to_native=lambda a: a.native,
-        )
-    if issubclass(typ, tuple) or typing.get_origin(typ) is tuple:
+    # For types that are expected to be simple classes for these specific checks
+    if inspect.isclass(typ):
+        if issubclass(typ, bool):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.Bool,
+                native_to_arc4=arc4.Bool,
+                arc4_to_native=lambda n: n.native,
+            )
+        if issubclass(typ, UInt64Backed):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.UInt64,
+                native_to_arc4=lambda n: arc4.UInt64(n.int_),
+                arc4_to_native=lambda a: typ.from_int(a.native),
+            )
+        if issubclass(typ, BigUInt):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.UInt512,
+                native_to_arc4=arc4.UInt512,
+                arc4_to_native=lambda a: a.native,
+            )
+        if issubclass(typ, Account):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.Address,
+                native_to_arc4=arc4.Address,
+                arc4_to_native=lambda a: a.native,
+            )
+        if issubclass(typ, UInt64):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.UInt64,
+                native_to_arc4=arc4.UInt64,
+                arc4_to_native=lambda a: a.native,
+            )
+        if issubclass(typ, Bytes):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.DynamicBytes,
+                native_to_arc4=arc4.DynamicBytes,
+                arc4_to_native=lambda a: a.native,
+            )
+        if issubclass(typ, String):
+            return _Serializer(
+                native_type=typ,
+                arc4_type=arc4.String,
+                native_to_arc4=arc4.String,
+                arc4_to_native=lambda a: a.native,
+            )
+
+    if origin_typ is tuple or (inspect.isclass(typ) and issubclass(typ, tuple)):
```

Comment:
Should we also add `inspect.isclass(typ)` clause to the next if statement on line 117 just to be safe?